pkg/virt-handler/vm.go

```diff
@@ -925,6 +926,23 @@ func (d *VirtualMachineController) execute(key string) error {
 	// Take different execution paths depending on the state of the migration and the
 	// node this is executed on.
 
+	if vmiExists && d.isMigrationSource(vmi) &&
```

Comment:
this would make more sense to me in the ```processVmUpdate()``` function as one of the "if else" statements involved with migrations there.

Reply:
I really don't mind moving the code around, but I don't really see a reason. Here is the place where all migration execution paths are consolidated, migration cancellation is one of them, from my pov.

Reply:
the way the logic at the moment is short-circuiting the vmi execute flow. These top level if statements are related to which execution path to take based on ownership of the VMI.

In this case, the defaultExecute path should be taken because the source node still owns the VMI. That function is determining all kinds of stuff related to whether or not the VMI is shutting down, crashed, watchdog file is alive, graceful shutdown, etc... Abort migration is a condition that should be handled as part of updating the current VMI if it is deemed that the VMI should be updated (follow the shouldUpdate boolean). 

This logic probably belongs somewhere around the same area in processVmUpdate that the migration is performed.

so this if statement

```
        } else if d.isMigrationSource(vmi) {                                                   
                err = client.MigrateVirtualMachine(vmi)                                        
                if err != nil {
                        return err
                }
                d.recorder.Event(vmi, k8sv1.EventTypeNormal, v1.Migrating.String(), "VirtualMachineInstance is migrating.")
        }
                
```

would become something like

```
        } else if d.isMigrationSource(vmi) {
                if vmi.Status.MigrationState.AbortRequested {
                    err = client.CancelVirtualMachineMigration(vmi)
                    if err != nil {
                            return err
                    }
                    d.recorder.Event(vmi, k8sv1.EventTypeNormal, v1.Migrating.String(), "VirtualMachineInstance is aborting migration.")
                } else { 
                    err = client.MigrateVirtualMachine(vmi)                                        
                    if err != nil {
                            return err
                    }
                    d.recorder.Event(vmi, k8sv1.EventTypeNormal, v1.Migrating.String(), "VirtualMachineInstance is migrating.")
                }

        }
```

Reply:
alright, updated.